types/event.go

```diff
@@ -119,7 +113,16 @@ func (t *Trace) AddSpan(sp *Span) {
 // GetSpans returns the list of spans in this trace
 func (t *Trace) GetSpans() []*Span {
 	return t.spans
+}
 
+// SetDeadline sets the deadline that this trace should be sent by
+// When we check for traces to send any trace that has exceeded the
+// deadline will be sent
+func (t *Trace) SetDeadline(d time.Duration) {
+	deadline := time.Now().Add(d)
+	if t.SendBy.IsZero() || t.SendBy.After(deadline) {
```

Comment:
This is an interesting test. I think it means "you're only allowed to set deadlines that are sooner, you can't push out when a trace should be sent".  Did I get that right? Do we need this protection? 

There's a future feature that might make us reconsider - instead of keeping a trace for 60s since the *first* span arrives, allow keeping a trace for 60s since the *most recent* span arrives. This is one approach to allowing very long traces (minutes, hours) while still getting them out the door at a reasonable time. We haven't accepted that work as a feature yet though - just flagging this as a thing that would have to change if we were.

```suggestion
	// only set the deadline if it has not yet been set or is sooner than
	// the existing deadline
	if t.SendBy.IsZero() || t.SendBy.After(deadline) {
```

Reply:
This is pretty much verbatim from the refinery code https://github.com/honeycombio/hound/blob/ced8efbf733fa4d7ea3b9b72ec677412a0651d51/cmd/dalmatian/app/collect/collect.go#L555. I think if we need to change this in the future then it's no big deal. I'll try to figure out why refinery uses this check though